### PR TITLE
[terraform] add cache layer to CI for modules and providers

### PR DESCRIPTION
**Pull request by yohanb**
**Description**

**Changes**

* feat(lint): add caching layer to terraform

🚀 PR created with [fotingo](https://github.com/tagoro9/fotingo)

### CODE REVIEW

**Review by tagoro9 — COMMENTED** [review]
Exciting! Just some thoughts about not using `$HOME`

File: lint/action.yaml [hunk 1]
```diff
@@ -37,6 +37,32 @@ runs:
     - name: Setup tools
       # Provide opportunity to install terraform, golang, and others
       uses: open-turo/action-setup-tools@v2
+
+    - name: Configure Terraform plugin cache
+      shell: bash
+      run: |
+        echo "TF_PLUGIN_CACHE_DIR=$HOME/.terraform.d/plugin-cache" >>"$GITHUB_ENV"
```
[the remark this answers is not in the document]
Comment (yohanb) [reply lint/action.yaml:44]: that folder is empty after each job.. wouldn't that defeat the purpose of the cache?

[the remark this answers is not in the document]
Comment (tagoro9) [reply lint/action.yaml:44]: I think it should be fine. You load from S3 into that folder before tf runs and you isolate runs between jobs of different repos as you are not using the runner home folders, which for not epehemeral runners is not cleaned up.

[the remark this answers is not in the document]
Comment (yohanb) [reply lint/action.yaml:44]: oh right... my bad 🤦

**Review by dalabi06 — APPROVED** [review]

**Review by tagoro9 — APPROVED** [review]

**Pull request merged**